ghost/core/core/server/services/collections/BookshelfCollectionsRepository.js

```diff
@@ -122,19 +132,68 @@ module.exports = class BookshelfCollectionsRepository {
             {id: data.id},
             {
                 require: false,
-                transacting: options.transaction
+                transacting: options.transaction,
+                withRelated: ['collectionPosts']
             }
         );
 
         if (!existing) {
             await this.#model.add(data, {
                 transacting: options.transaction
             });
+            const thingsToInsert = collection.posts.map((postId, index) => {
+                return {
+                    id: (new ObjectID).toHexString(),
+                    sort_order: collection.type === 'manual' ? index : 0,
+                    collection_id: collection.id,
+                    post_id: postId
+                };
+            });
+            if (thingsToInsert.length > 0) {
+                await this.#relationModel.query().insert(thingsToInsert).transacting(options.transaction);
+            }
         } else {
-            return this.#model.edit(data, {
+            await this.#model.edit(data, {
                 id: data.id,
                 transacting: options.transaction
             });
+
+            const things = collection.posts.map((postId, index) => {
+                return {
+                    id: (new ObjectID).toHexString(),
+                    sort_order: collection.type === 'manual' ? index : 0,
+                    collection_id: collection.id,
+                    post_id: postId
+                };
+            });
+
+            const thingsToDelete = [];
+
+            if (collection.type === 'manual') {
+                await this.#relationModel.query().delete().where('collection_id', collection.id).transacting(options.transaction);
+            } else {
+                const existingRelations = existing.toJSON().collectionPosts;
+
+                for (const existingRelation of existingRelations) {
+                    const found = things.find((thing) => {
+                        return thing.post_id === existingRelation.post_id;
+                    });
+                    if (found) {
+                        found.id = null;
+                    } else {
+                        thingsToDelete.push(existingRelation.id);
+                    }
+                }
+            }
+
+            const thingsToInsert = things.filter(thing => thing.id !== null);
+
+            if (thingsToInsert.length > 0) {
+                await this.#relationModel.query().insert(thingsToInsert).transacting(options.transaction);
+            }
+            if (thingsToDelete.length > 0) {
+                await this.#relationModel.query().delete().whereIn('id', thingsToDelete).transacting(options.transaction);
```

Comment:
Vaguely reminds of the problems we used to deal with in SQLite where the "WHERE IN (...)" was not performing due to a [query parameter limit](https://github.com/TryGhost/Ghost/issues/12032). Good those days are over 😅